src/Symfony/Component/Messenger/Tests/Transport/AmqpExt/ConnectionTest.php

```diff
@@ -275,6 +275,34 @@ public function testItSetupsTheConnection()
         $connection->publish('body');
     }
 
+    public function testBindingArguments()
+    {
+        $amqpConnection = $this->createMock(\AMQPConnection::class);
+        $amqpChannel = $this->createMock(\AMQPChannel::class);
+        $amqpExchange = $this->createMock(\AMQPExchange::class);
+        $amqpQueue = $this->createMock(\AMQPQueue::class);
+
+        $factory = $this->createMock(AmqpFactory::class);
+        $factory->method('createConnection')->willReturn($amqpConnection);
+        $factory->method('createChannel')->willReturn($amqpChannel);
+        $factory->method('createExchange')->willReturn($amqpExchange);
+        $factory->method('createQueue')->willReturn($amqpQueue);
+
+        $amqpExchange->expects($this->once())->method('declareExchange');
+        $amqpExchange->expects($this->once())->method('publish')->with('body', 'routing_key', AMQP_NOPARAM, ['headers' => []]);
+        $amqpQueue->expects($this->once())->method('declareQueue');
+        $amqpQueue->expects($this->exactly(1))->method('bind')->withConsecutive(
+            [self::DEFAULT_EXCHANGE_NAME, null, ['x-match' => 'all']]
+        );
+
+        $dsn = 'amqp://localhost?'.
+            'exchange[default_publish_routing_key]=routing_key&'.
+            'queues[queue0][binding_arguments][x-match]=all';
```

Comment:
x-match is for headers exchanges. but you are not defining a headers exchange. please make the test as real as possible even if mocked because otherwise it doesn't make sense for the reader.

Reply:
Never had to use the queue binding arguments to be honest, but that's my understanding from [this article](https://www.cloudamqp.com/blog/2015-09-03-part4-rabbitmq-for-beginners-exchanges-routing-keys-bindings.html):
> A special argument named "x-match", which can be added in the binding between your exchange and your queue, tells if all headers must match or just one

Reply:
as I said x-match only applies for HEADERS exchange. you are using the default direct exchange. you need to change the exchange type.

Reply:
@fabpot updated as per @Tobion's comments 👍